src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,48 +1,153 @@
 package core.basesyntax;
 
+import java.util.NoSuchElementException;
+
 public class ArrayList<T> implements List<T> {
+    private static final int DEFAULT_CAPACITY = 10;
+    private static final int NO_INDEX = -1;
+    private static final double DEFAULT_LOAD_FACTOR = 1.5;
+    private int size;
+    private int currentCapacity = DEFAULT_CAPACITY;
+    private T[] elementData;
+
+    public ArrayList() {
+        elementData = (T[]) new Object[DEFAULT_CAPACITY];
+    }
+
     @Override
     public void add(T value) {
-
+        elementData[size] = value;
+        size++;
+        growIfArrayIsFull();
     }
 
     @Override
     public void add(T value, int index) {
-
+        if (index > size || index < 0) {
+            throw new ArrayListIndexOutOfBoundsException(
```

Comment:
Consider using constants for error messages to avoid duplication and to make it easier to manage them. For example, you could define a constant like `INDEX_OUT_OF_BOUNDS_EXCEPTION_MESSAGE` and use it throughout your class.